HexEditor: Use `Window::set_modified()`

krkk opened the pull request:
This commit will show a special close icon in the window frame when a file is modified.

![screenshot (1,2K)](https://user-images.githubusercontent.com/16520278/154026794-2860d818-3b0f-4e78-a1e5-a2128d3a9100.png)

trflynn89 commented:
The commit title leaves a bit to be desired, `Use <function_name>` isn't particularly descriptive. How about `HexEditor: Display dirty close icon when the file is changed` or something in the vein?

(Same comment about #12522)

krkk commented:
Changes since last push:
- reworded the commit
- rebased to master
- fixed a bug from the previous patch where the window wouldn’t close after clicking “Yes”. oops.
